### PR TITLE
fastqc-nf-test : mv outputdir param to test file and add snapshot

### PR DESCRIPTION
**Pull request by sateeshperi**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/fastqc/tests/main.nf.test [hunk 1]
```diff
@@ -21,12 +25,12 @@ nextflow_process {
...
         then {
-            assert process.success
-            assert process.out.html.get(0).get(1) ==~ ".*/test_fastqc.html"
-            assert path(process.out.html.get(0).get(1)).getText().contains("<tr><td>File type</td><td>Conventional base calls</td></tr>")
-            assert process.out.zip.get(0).get(1) ==~ ".*/test_fastqc.zip"
+            assertAll (
+            { assert process.success },
+            { assert snapshot(process.out.html).match("html") },
+            { assert snapshot(process.out.versions).match("versions") },
+            { assert process.out.zip.get(0).get(1) ==~ ".*/test_fastqc.zip" }
```
Comment (SPPearce) [inline modules/nf-core/fastqc/tests/main.nf.test:32]: What does this do?

Comment (sateeshperi) [reply modules/nf-core/fastqc/tests/main.nf.test:32]: 1. check if the process has succeeded
2. insert process out `html` into snapshot
3. 2. insert process out `versions` into snapshot 
4. check if the process out `zip` produces a zip file

**Review by SPPearce — APPROVED** [review]
Ok, looks good to me then.

File: modules/nf-core/fastqc/tests/main.nf.test [hunk 2]
```diff
@@ -21,12 +25,12 @@ nextflow_process {
         }
 
         then {
-            assert process.success
-            assert process.out.html.get(0).get(1) ==~ ".*/test_fastqc.html"
-            assert path(process.out.html.get(0).get(1)).getText().contains("<tr><td>File type</td><td>Conventional base calls</td></tr>")
-            assert process.out.zip.get(0).get(1) ==~ ".*/test_fastqc.zip"
+            assertAll (
+            { assert process.success },
+            { assert snapshot(process.out.html).match("html") },
```
Comment (Emiller88) [inline modules/nf-core/fastqc/tests/main.nf.test:30]: I don't think you can md5sum the html report

Comment (Emiller88) [reply modules/nf-core/fastqc/tests/main.nf.test:30]: That's so weird. Why did we never test it with pytest-workflow?

Oh well, good enough for me!

**Comment by SPPearce** [comment]
> Just nf-test magic I guess 🤷🏻

Nope, I remember this one, the report contains the date inside it, which means that the md5sum is stable per day, but not longer than that. So you can't md5sum it.

**Comment by SPPearce** [comment]
Yeah, looks like this:
`<div id="header_filename">Mon 2 Oct 2023<br/>test.gz</div>`

**Review by SPPearce — APPROVED** [review]
Heh, very explicit comment there ;)

**Pull request merged**